src/codegen/llvm/codegen_llvm_visitor.cpp

```diff
@@ -28,12 +32,31 @@ static constexpr const char instance_struct_type_name[] = "__instance_var__type"
 /*                            Helper routines                                           */
 /****************************************************************************************/
 
+/// A utility to check for supported Statement AST nodes.
 static bool is_supported_statement(const ast::Statement& statement) {
     return statement.is_codegen_var_list_statement() || statement.is_expression_statement() ||
            statement.is_codegen_for_statement() || statement.is_codegen_return_statement() ||
            statement.is_if_statement() || statement.is_while_statement();
 }
 
+/// A utility to check of the kernel body can be vectorised.
+static bool can_vectorise(const ast::CodegenForStatement& statement, symtab::SymbolTable* sym_tab) {
+    // Check that function calls are made to external methods only.
```

Comment:
Just a note - this will be expanded bit further in future but don't hesitate to comment out certain constructs from DERIVATIVE blocks if you see something not compatible :) (e.g. `if statements` are ok to comment for benchmarking purpose)

Reply:
Yes! This is just a useful precondition to catch unhandled cases rather than figure out what went wrong in IR:) (If not commented out)